diplo.lua

```diff
@@ -0,0 +1,88 @@
+--[[
+diplo - Quick "Diplomacy"
+Without arguments:
+    Print list of civs the player civ has diplomatic relations to.
+    This list might not match the ingame civ list, but is what matters. Does not include 'no contact' civs, etc.
+    Also shows if relations match between player civ and other civ - a mismatch can potentially happen 'naturally', but is important when trying to make peace proper, as both civs need to be at peace with the other.
+With arguments:
+    diplo CIV_ID RELATION
+    Changes the relation to the civ identified by CIV_ID to the one specified in RELATION, making sure that this is mutual.
+    Relations: 0 = Peace; 1 = War; 3 = Alliance (seems to only apply to site govs...?)
+    Civ list is shown afterwards.
+]]
+
+local args = {...}
+
+-- player civ references:
+local p_civ_id = df.global.plotinfo.civ_id
+local p_civ = df.historical_entity.find(df.global.plotinfo.civ_id)
+
+-- get a very rough, but readable name for a civ:
+function get_raw_name(civ)
+    local raw_name = ""
+    for _, name_word in pairs(civ.name.words) do
+        if name_word ~= -1 then
+            raw_name = raw_name .. " " .. df.global.world.raws.language.words[name_word].word
+        end
+    end
+    raw_name = string.sub(raw_name, 2)

```

Comment:
what does this chop off?

Reply:
Leading whitespace on the very rudimentarily built names. Is there a dfhack utility that returns the properly inflected/built names? Currently they come out of this like 'COOPERATE CHANNEL' for 'The Cooperative Channel'.

Reply:
I'd bring that question to the DFHack discord. More people are familiar with languages handling there

Reply:
Using dfhack.TranslateName now.